Robust_Lab.qmd

```diff
@@ -74,11 +187,62 @@ Use the `stars` data in `robustbase`. This data looks at the relationship betwee
 
     f\. Plot bootstrap mean along with 95% CIs (with `ggplot2`). Use annotate to note what the vertical lines represent.
 
+    ```{r}
+    #a
+    pre <- c(22,25,17,24,16,29,20,23,19,20)
+    post <- c(18,21,16,22,19,24,17,21,23,18)
+    dataQ3 <- data.frame(pre, post)
+    pdiff0 <- (dataQ3$post - dataQ3$pre)
+
+    #b
+    Xpdiff0 <-mean(pdiff0)
+    Xpdiff0
+
+    #d
+    mean_fun1 = function(pdiff0, indices) {
+      return(mean(pdiff0[indices])) 
+    results3 = boot(pdiff0, mean_fun1, R=1000) }
+    pdiff1=results3$t
+    results3
+    hist(pdiff1)
+
+    #e
+    results_ci_bcaQ3 = boot.ci(results3, type = "bca", R=1000)
+    results_ci_bcaQ3
+    #Inference: The CI includes zero, so there is not a statistically significant difference between the two groups.
+
+    #f
+    hist(pdiff1)
+    abline(v=c(-3.100, .464), col=c("blue", "blue"), lty=c(1,2, 1,2), lwd=c(1, 1, 3,3))
+    ```
+
 4.  Pepper Joe measured the length and heat of 85 chili peppers. He wants to know if smaller peppers are hotter than longer peppers.
 
     ```{r}
     #read data.table to read in
-    chili<- read.delim("/Users/jasongeller/Documents/03-Robust_Methods/data/chillis.csv")
+    chili<- read.delim("https://raw.githubusercontent.com/jgeller112/psy504-advanced-stats/main/slides/03-Robust_Methods/data/chillis.csv")
+    ```
+
+    ```{r}
+
```

Comment:
we are essentially bootrapping the coefs from the regression model. Something like below would work: 

boot.function <- function(data, indices) {
resampled_data = data[indices,]
model<-glm(HEAT~LENGTH, data=resampled_data)
return(coefficients(model))}

results <- boot(chili, boot.function, R=10000)

boot.ci(results, type="bca", R=10000)

Reply:
You are just bootstrapping the mean which would not allow us to answer the question.